src/components/TextBoxField/index.tsx

```diff
@@ -0,0 +1,175 @@
+import type * as SCWRegexType from '@scaleway/regex'
+import { TextBox } from '@scaleway/ui'
+import { FieldState } from 'final-form'
+import React, { Ref, forwardRef, useMemo } from 'react'
+import { useField, useFormState } from 'react-final-form'
+import pickValidators from '../../helpers/pickValidators'
+import useValidation from '../../hooks/useValidation'
+import { useErrors } from '../../providers/ErrorContext'
+import { BaseFieldProps } from '../../types'
+
+export type TextBoxFieldProps<T = unknown, K = string> = BaseFieldProps<
+  T,
+  K
+> & {
+  label?: string
+  name: string
+  placeholder?: string
+  required?: boolean
+  min?: number
+  max?: number
+  minLength?: number
+  maxLength?: number
+  regex?: keyof typeof SCWRegexType
```

Comment:
What if we want multiple regex ?

Reply:
How can this is possible ? regex is a prop to use one regex of scaleway/regex and we can keep pattern prop. Otherwise the user can pass a validate prop ?

Reply:
Currently we have this in rdb create for example

```tsx
  const validateName = useValidators(
    mustHaveFirstCharacterAlpha,
    mustBeAlphanumericUnderScoresDollarsDashes,
  )
```